--- a/x.c
+++ b/x.c
@@ -1196,8 +1196,8 @@ xhints(void)
 	sizeh->height_inc = win.ch;
 	sizeh->width_inc = win.cw;
 	sizeh->base_height = 2 * borderpx;
 	sizeh->base_width = 2 * borderpx;
-	sizeh->min_height = win.ch + 2 * borderpx;
-	sizeh->min_width = win.cw + 2 * borderpx;
+	sizeh->min_height = 24 * win.ch + 2 * borderpx;  /* min 24 rows */
+	sizeh->min_width  = 40 * win.cw + 2 * borderpx;  /* min 40 cols */
 	if (xw.isfixed) {
 		sizeh->flags |= PMaxSize;
 		sizeh->min_width = sizeh->max_width = win.w;
